src/contact.rs

```diff
@@ -675,20 +678,59 @@ impl Contact {
     }
 
     /// Get blocked contacts.
-    pub async fn get_all_blocked(context: &Context) -> Vec<u32> {
-        context
+    pub async fn get_all_blocked(context: &Context) -> Result<Vec<u32>> {
+        // add blocked mailinglists as contacts
+        // to allow unblocking them as if they are contacts
+        // (this way, only one unblock-ffi is needed and only one set of ui-functions,
+        // from the users perspective,
+        // there is not much difference in an email- and a mailinglist-address)
+        let blocked_mailinglists = context
```

Comment:
I would rather have the new code as a separate function (`update_blocked_mailinglist_contacts`) or something like this, and call it at the top of `get_all_blocked()`. Otherwise it is hard to find where the actual logic of getting the contacts from the database starts without reading the whole function and realizing only a few last lines actually do the job.

Reply:
+1

Reply:
this makes also error handling better, if `update_blocked_mailinglist_contacts` fails for some reason, `get_all_blocked` should still return the list of blocked contacts.

Reply:
moved it to a function as suggested, much better, indeed!